### PR TITLE
VFS-783 - Based on WEBDAV4 vs WEBDAV4S set backend http scheme 

### PR DESCRIPTION
**Pull request by satish-csi**
VFS-783 - Based on WEBDAV4 vs WEBDAV4S set backend http scheme

### CODE REVIEW

**Review by michael-o — APPROVED** [review]

File: commons-vfs2-jackrabbit2/src/main/java/org/apache/commons/vfs2/provider/webdav4/Webdav4FileObject.java [hunk 1]
```diff
@@ -544,7 +544,8 @@ DavPropertySet getPropertyNames(final GenericURLFileName name) throws FileSystem
      * @return The encoded URL String.
      */
     private String hrefString(final GenericURLFileName name) {
-        final GenericURLFileName newFile = new GenericURLFileName("http", name.getHostName(), name.getPort(), name.getDefaultPort(),
+        final String scheme = getHttpScheme(name);
```
Comment (woonsan) [inline commons-vfs2-jackrabbit2/src/main/java/org/apache/commons/vfs2/provider/webdav4/Webdav4FileObject.java:547]: `#getHttpScheme(name)` is unnecessary if you just use `getInternalURI().getScheme()` and move the line into try ~ catch block.

Comment (satish-csi) [reply commons-vfs2-jackrabbit2/src/main/java/org/apache/commons/vfs2/provider/webdav4/Webdav4FileObject.java]: Hi @woonsan ,
Thanks for suggestions! I have updated code as per your suggestion! please review again!

**Comment by garydgregory** [comment]
@satish-csi Needs unit tests.

**Comment by woonsan** [comment]
Hi @garydgregory ,
I've cherry-picked @satish-csi 's commits and just added a unit test class to verify the fix in another PR: https://github.com/apache/commons-vfs/pull/143
Could you please review it? Thanks.

**Pull request merged**